pkg/controller/association/conf.go

```diff
@@ -103,6 +107,38 @@ func ElasticsearchAuthSettings(c k8s.Client, association commonv1.Association) (
 	return assocConf.AuthSecretKey, string(data), nil
 }
 
+// AllowVersion returns true if the given resourceVersion is lower or equal to the associations version.
+// For example: Kibana in version 7.8.0 cannot be deployed if its Elasticsearch association reports version 7.7.0.
+// Referenced resources version is parsed from the association conf annotation.
+func AllowVersion(resourceVersion version.Version, associated commonv1.Associated, logger logr.Logger) bool {
+	for _, assoc := range associated.GetAssociations() {
+		assocRef := assoc.AssociationRef()
+		if !assocRef.IsDefined() {
+			// no association specified, move on
+			continue
+		}
+		if assoc.AssociationConf() == nil || assoc.AssociationConf().Version == "" {
+			// no conf reported yet, this may be the initial resource creation
+			logger.Info("Delaying version deployment since the version of an associated resource is not reported yet",
+				"version", resourceVersion, "ref_namespace", assocRef.Namespace, "ref_name", assocRef.Name)
+			return false
+		}
+		refVer, err := version.Parse(assoc.AssociationConf().Version)
+		if err != nil {
+			logger.Error(err, "Invalid version found in association conf", "association_version", refVer)
+			return false
+		}
+		if !refVer.IsSameOrAfter(resourceVersion) {
+			// the version of the referenced resource (example: Elasticsearch) is lower than
+			// the desired version of the reconciled resource (example: Kibana)
+			logger.Info("Delaying version deployment since an associated resource is not upgraded yet",
```

Comment:
I think we should we also post events here too. If the user just applied higher version on ref holder, then nothing is going to happen. I think the first place they're going to look at for explanation is resource events.

Reply:
https://github.com/elastic/cloud-on-k8s/issues/3538 should also help, but I agree. I'll add the event 👍